### ISSUE TITLE
LuckPerms Bungee Prefix

### ISSUE THREAD

**Issue by xtechgamer735** [lucko/LuckPerms#817]
Hello!

I am trying to make it so the bungee version of LuckPerms displays the correct prefix dependant on what server the player is currently connected to. I want it to show in the "/lpb user <user> info section". The ideas is to get this info into my own chat plugin eventually but until LuckPerms displays it correctly internally I suppose I've got no chance. Im using the code given in the wiki to get that value, but currently using the context of "allow all" as a workaround. It doesn't update when they change server so has prison ranks on creative for eg.

For example if they are on server 'skyprison' it should display the prefix which is attached to group 'fire' which is '[Fire]' (priority 2) that is assigned to them with context server 'skyprison'. I am unable to make it do this and give up myself.

Currently, the user has default group with a global prefix [Guest] (priority 0), plus a secondary prefix that is server specific for server 'skyprison' that is [G] (priority 1), to follow server tradition. That is the prefix that is showing currently. The problem is that the user has the fire parent assigned with context 'skyprison' and it refuses to show its meta even though the meta has a) a higher weight and b) correctly set for server bungee (name of LPB instance) and world set to the server name 'skyprison'

As you can see from the below screen shots, I have assigned server and world settings correctly for a bungee setup.

Any pointers will be much appreciated as to how I can do this,

Many thanks,

/lpb user <user> info = https://imgur.com/a/xpd8l
/lpb user <user> meta info = https://imgur.com/a/piPAh
Prefix Setting = https://imgur.com/a/tuj45

**Comment by VulpusMaximus** [comment]
https://github.com/lucko/LuckPerms/wiki/Command-Usage:-Meta#lp-usergroup-usergroup-meta-addprefix

**Comment by VulpusMaximus** [comment]
So, for you it would be something like /lp group fire 2  "[Fire]" sky

**Comment by xtechgamer735** [comment]
Hello! As far as I am aware this is set correctly, as advised by discord support.

**Comment by BrainStone** [comment]
For the Bungee different servers are considered worlds

**Comment by xtechgamer735** [comment]
Hi Brain, as you will see from the above screen shots this is how I’ve set it up. With every prefix assigned to the ‘bungee’ server with the correct server name as the world context. Many thanks!

**Comment by VulpusMaximus** [comment]
They should be set to the server skyprison, world global

**Comment by VulpusMaximus** [comment]
Not server bungee

**Comment by xtechgamer735** [comment]
As far as I understand, the bungee version handles servers and worlds hence this setup. It was originally setup as you suggest and it didn’t work.

**Comment by lucko** [comment]
Looks fine to me.

What chat plugin are you using? This'll only work if you have a BungeeCord chat plugin installed. 

As the correct value is being shown in the`info` command output - it's most likely not a LP issue.

**Comment by xtechgamer735** [comment]
Hey Luck,

Thanks for your reply. The problem is that sadly the prefix is not being shown correctly inside luck perms.

As you will see from the screenshots, the user has the rank and meta for 'fire' applied but luck perms shows the default and lowest priority prefix for that user in `info`.

**Comment by lucko** [comment]
Can you send me the output to `/lpb debug` please

**Comment by xtechgamer735** [comment]
It gives me an NPE when I try to run that command https://pastebin.com/NFYafVse

Using 4.1.25 :)

**Comment by lucko** [comment]
Update to the latest
https://ci.lucko.me/job/LuckPerms/

**Comment by xtechgamer735** [comment]
No luck either sadly https://pastebin.com/fn2P3jKd

**Comment by xtechgamer735** [comment]
https://hastebin.com/raw/evehemamuc got it :)

**Comment by lucko** [comment]
Ok thanks, I'm going to try to reproduce this tomorrow (going to sleep now)

**Comment by xtechgamer735** [comment]
Thank you very much! Please give me a shout if you need anything else. I don't blame you, me too :)

**Comment by lucko** [comment]
Can you send me an export file for your groups?

Run: `/lp export export.txt` and then send me the contents of `export.txt` in the LuckPerms folder

**Comment by xtechgamer735** [comment]
[export.txt](https://github.com/lucko/LuckPerms/files/1812424/export.txt)

**Comment by lucko** [comment]
Ha okay, I got this working locally and immediately found the problem. 😄 

`subzeronetwork` only inherits from `fire` in `server=skyprison` and importantly **not** on `server=bungee`.

Without this inheritance, their meta info looks like this.

![](https://luck.sh/Um7ilcellVgRKYIFkzQ9wTfCd_s=)

Which is where the [G] prefix comes from.

The meta info output is a bit confusing I admit

![](https://luck.sh/Jz9mkSN5rl7MPtA3GVL24WpLu2w=)

Each of those lines should infact say something like..

```
-> 2 - '[Fire]' (inherited from fire on server=skyprison) (server=bungee) (world=skyprison)
```

tl;dr you need to have your inheritances apply on `server=bungee` too.

**Comment by xtechgamer735** [comment]
Ah I think I understand, many thanks for looking into this. Is it possible to automate this in any way, so everything has the context of bungee automatically?

Groups change regularly you see and would be handy if this could be the case.

Many thanks.

**Comment by lucko** [comment]
It's possible via the API using the [`FullySatisfiedContexts`](https://github.com/lucko/LuckPerms/blob/master/api/src/main/java/me/lucko/luckperms/api/FullySatisfiedContexts.java) instance.

It's a bit of a hack - I'm not sure if it would ever make its way into the config file.

**Comment by xtechgamer735** [comment]
Ooh that sounds up my street, I plan on using the API for my chat plugin.

Do you happen to have an example of it knocking around anywhere? No probs if not :)

**Comment by lucko** [comment]
Yes, there are lots of examples & explanations here:

https://github.com/lucko/LuckPerms/wiki/Developer-API
https://github.com/lucko/LuckPerms/wiki/Developer-API:-Usage

**Comment by xtechgamer735** [comment]
Lovely stuff thanks! Thank you very much for your help with this :)
